### PR TITLE
Add -pthread if pthreads enabled (Issue #2363)

### PR DESCRIPTION
**Pull request by jeffmiles63**
Issue #2363

### CODE REVIEW

File: cmake/kokkos_arch.cmake [hunk 1]
```diff
@@ -448,6 +448,9 @@ UNSET(_HOST_PARALLEL)
 
 IF(KOKKOS_ENABLE_PTHREAD)
   SET(KOKKOS_ENABLE_THREADS ON)
+  ARCH_FLAGS(
```
Comment (crtrott) [inline cmake/kokkos_arch.cmake]: @jjwilke is this the right place to add it?

Comment (jjwilke) [reply cmake/kokkos_arch.cmake]: It's probably not an architecture flag. I would probably put it in `kokkos_enable_devices.cmake` around line 26 where there's already a pthread if-block.

I started the compiler options as global variables... so leave them that way for now. They can probably become regular variables. Anyhow, the relevant code is:

````
  LIST(APPEND KOKKOS_COMPILE_OPTIONS -pthread)
  LIST(APPEND KOKKOS_LINK_OPTIONS -pthread)
  GLOBAL_SET(KOKKOS_COMPILE_OPTIONS ${KOKKOS_COMPILE_OPTIONS})
  GLOBAL_SET(KOKKOS_LINK_OPTIONS ${KOKKOS_LINK_OPTIONS})
````

I wasn't sure how the Trilinos stuff was going to shake out so I made several things globals that weren't necessary to be globals. If we convert these to regular variables, you can just append:

````
  LIST(APPEND KOKKOS_COMPILE_OPTIONS -pthread)
  LIST(APPEND KOKKOS_LINK_OPTIONS -pthread)
````

Comment (crtrott) [reply cmake/kokkos_arch.cmake]: @jjwilke can you make a concrete counter proposal for this?

**Comment by junghans** [comment]
Here is a more CMakeish solution to the problem: https://github.com/kokkos/kokkos/pull/2405

**Comment by jeffmiles63** [comment]
closing in lieu of #2405

**Pull request closed without merging**